### ISSUE TITLE
.NET 6: Alpine backporting

### ISSUE THREAD

**Issue by ayakael** [dotnet/source-build#3087]
# Issue tracker for Alpine patch backporting for .NET 6

## Goals
Backport all patches currently used by Alpine's  [dotnet6](https://gitlab.alpinelinux.org/alpine/aports/-/tree/master/community/dotnet6-build) package 

### Clang patches
- [x] runtime_73065-define-cpuidex-only-when-no-builtin.patch 
https://github.com/dotnet/runtime/pull/77507

### Source-build related patches
- [ ] sdk_14239-add-zsh-compdef-completion-script.patch 
(waiting on main)
- [x] sdk_22373-portable-runtime-id.patch
https://github.com/dotnet/sdk/pull/27849

### s390x / ppc64le related patches
- [x] msbuild_optional-systemconfiguration.patch 
(not to be backported)
- [x] msbuild_optional-systemsecurity.patch 
(not to be backported)
- [ ] roslyn_57003-mono-namedmutex.patch 
(to be backported)
- [ ] runtime_76500-mono-musl-support.patch 
https://github.com/dotnet/runtime/pull/78274
- [x] runtime_60315-momo-remove-ilstrip.patch 
https://github.com/dotnet/runtime/pull/77503
- [x] runtime_60675-mono-sterror.patch 
https://github.com/dotnet/runtime/pull/77505
- [x] runtime_66594-keep-native-symbols.patch 
https://github.com/dotnet/runtime/pull/77506
- [ ] arcade_11608-crossbuild-fixes.patch
(waiting on main)

### arm related patches
- [x] installer_13378-arm-build.patch 
https://github.com/dotnet/installer/pull/13378

### Non-portable build patches
- [x] aspnetcore_43937-support-building-with-nonportable-runtime.patch 
(https://github.com/dotnet/aspnetcore/pull/44752)
- [ ] build_14549-no-portable-build 
https://github.com/dotnet/installer/pull/14816
- [x] runtime_74504-pass-targetrid-to-native-scripts.patch 
https://github.com/dotnet/runtime/pull/77508
https://github.com/dotnet/arcade/pull/11604
https://github.com/dotnet/runtime/pull/78425
- [x] runtime_75597-support-building-runtime-with-non-portable-runtime.patch 
https://github.com/dotnet/runtime/pull/77509
- [x] runtime_76068-use-generated-runtimejson-when-building-sharedframework.patch 
https://github.com/dotnet/runtime/pull/77510
- [x] sdk_28380-map-nonportable-rids-when-targetos-is-determined.patch
https://github.com/dotnet/sdk/pull/28782

**Comment by ayakael** [comment]
@omajid I went ahead and made a PR for removing ILStrip dependency on Mono, seeing as how it doesn't seem like you've upstreamed it: https://github.com/dotnet/runtime/pull/77503

also, for should I make a PR for the msbuild stuff?

Finally, for roslyn, how do I go about backporting the patch given lack of `release/6.0` branch?

**Comment by omajid** [comment]
> Also, should I make a PR for the msbuild stuff?

See https://github.com/dotnet/source-build/issues/2618#issuecomment-1290858035. We could continue carrying this patch locally, but sharing it would be better.

> Finally, for roslyn, how do I go about backporting the patch given lack of `release/6.0` branch?

That's a good question. If we look in dotnet/installer's `release/6.0.1xx` branch, it's using roslyn commit 487283bcd8d66693091f2800dcf1c8ae37cccdee: https://github.com/dotnet/installer/blob/5700aea798f6940ad4a713b0bc3b31bdf6e8efd8/eng/Version.Details.xml#L144-L148

```
$ cd roslyn
$ git branch --contains 487283bcd8d66693091f2800dcf1c8ae37cccdee
* main
  release/dev17.4
```

So, it sounds like we should make the PR against the `release/dev17.4` branch? Hopefully a roslyn maintainer can correct us if we get it wrong.

**Comment by ayakael** [comment]
> > Also, should I make a PR for the msbuild stuff?
> 
> See [#2618 (comment)](https://github.com/dotnet/source-build/issues/2618#issuecomment-1290858035). We could continue carrying this patch locally, but sharing it would be better.
> 
> > Finally, for roslyn, how do I go about backporting the patch given lack of `release/6.0` branch?
> 
> That's a good question. If we look in dotnet/installer's `release/6.0.1xx` branch, it's using roslyn commit 487283bcd8d66693091f2800dcf1c8ae37cccdee: https://github.com/dotnet/installer/blob/5700aea798f6940ad4a713b0bc3b31bdf6e8efd8/eng/Version.Details.xml#L144-L148
> 
> ```
> $ cd roslyn
> $ git branch --contains 487283bcd8d66693091f2800dcf1c8ae37cccdee
> * main
>   release/dev17.4
> ```
> 
> So, it sounds like we should make the PR against the `release/dev17.4` branch? Hopefully a roslyn maintainer can correct us if we get it wrong.

Ulrich beat me to the punch, they backported the patch a week ago. Thanks @uweigand!

**Comment by ayakael** [comment]
Actually, what am I talking about! They fixed the bug a week ago + 1 year. I think we need to update our `Version.Details.Xml` to pull latest commit from `dev/17.4`

**Comment by ayakael** [comment]
`dev/17.4` is way too new for `release/6.0.1xx`, and there's no branch dedicated to `release/6.0.1xx`. @MichaelSimons Is this something that we can propose to the roslyn devs? I see they have a `release/6.0.1xx-rc2` branch. It'd be nice to have the option to maintain servicing branches in the future.

If not, is this something we could ship within the `source-build` patches for `release/6.0.1xx` of installer?

**Comment by ayakael** [comment]
Would anyone know when we will jump to a new feature version (like 6.0.4xx) so I can get ahead of the curve and backport the proper patches to avoid regressions?

**Comment by MichaelSimons** [comment]
> Would anyone know when we will jump to a new feature version (like 6.0.4xx) so I can get ahead of the curve and backport the proper patches to avoid regressions?

Source-build will only support the 6.0.1xx feature band.  This is documented [here](https://github.com/dotnet/source-build/#support).  FWIW, we are working on supporting all feature bands in the future.

**Comment by ayakael** [comment]
> > Would anyone know when we will jump to a new feature version (like 6.0.4xx) so I can get ahead of the curve and backport the proper patches to avoid regressions?
> 
> Source-build will only support the 6.0.1xx feature band. This is documented [here](https://github.com/dotnet/source-build/#support). FWIW, we are working on supporting all feature bands in the future.

Copy that, so for .NET 8 we should expect all feature bands to be supported?

**Comment by ayakael** [comment]
Considering https://github.com/dotnet/roslyn/issues/65174, roslyn dev teams won't be creating a branch for SDK releases. Can we then integrate the roslyn patch within source-build?

**Comment by ayakael** [comment]
Closed now that patched are mostly backported
